src/platform_impl/linux/global_shortcut.rs

```diff
@@ -1,389 +1,110 @@
-              }
-            }
-            Ok(HotkeyMessage::DropThread) => {
-              (xlib.XCloseDisplay)(display);
-              return;
-            }
-            Err(err) => {
-              if let TryRecvError::Disconnected = err {
-                #[cfg(debug_assertions)]
-                eprintln!("hotkey: try_recv error {}", err);
-              }
-            }
-            _ => unreachable!("other message should not arrive"),
-          };
-
-          std::thread::sleep(std::time::Duration::from_millis(50));
-        }
-      }
-    });
-
-    ShortcutManager {
-      shortcuts: hotkeys,
-      method_sender,
-      method_receiver,
+  pub(crate) fn new<T>(window_target: &EventLoopWindowTarget<T>) -> Self {
+    Self {
+      tx: window_target.p.shortcut_requests_tx.clone(),
     }
   }
 
   pub(crate) fn register(
     &mut self,
     accelerator: Accelerator,
   ) -> Result<RootGlobalShortcut, ShortcutManagerError> {
-    let keycode = get_x11_scancode_from_hotkey(accelerator.key);
-
-    if let Some(keycode) = keycode {
-      let mut converted_modifiers: u32 = 0;
-      if accelerator.mods.shift_key() {
-        converted_modifiers |= xlib::ShiftMask;
-      }
-      if accelerator.mods.super_key() {
-        converted_modifiers |= xlib::Mod4Mask;
-      }
-      if accelerator.mods.alt_key() {
-        converted_modifiers |= xlib::Mod1Mask;
+    if let Some(key) = get_gtk_key(&accelerator) {
+      let id = accelerator.clone().id();
+      if let Err(e) = self.tx.send(ShortcutEvent::Register((id, key))) {
+        log::warn!(
+          "Failed to send global shortcut event to event loop channel: {}",
+          e
+        );
       }
-      if accelerator.mods.control_key() {
-        converted_modifiers |= xlib::ControlMask;
-      }
-
-      self
-        .method_sender
-        .send(HotkeyMessage::RegisterHotkey(
-          (0, 0),
-          converted_modifiers,
-          keycode,
-        ))
-        .map_err(|_| {
-          ShortcutManagerError::InvalidAccelerator("Unable to register global shortcut".into())
-        })?;
-
-      return match self.method_receiver.recv() {
-        Ok(HotkeyMessage::RegisterHotkeyResult(Ok(id))) => {
-          self
-            .shortcuts
-            .lock()
-            .unwrap()
-            .insert(id, accelerator.clone().id().0 as u32);
-          let shortcut = GlobalShortcut { accelerator };
-          return Ok(RootGlobalShortcut(shortcut));
-        }
-        Ok(HotkeyMessage::RegisterHotkeyResult(Err(err))) => Err(err),
-        Err(err) => Err(ShortcutManagerError::InvalidAccelerator(err.to_string())),
-        _ => Err(ShortcutManagerError::InvalidAccelerator(
-          "Unknown error".into(),
-        )),
-      };
+      Ok(RootGlobalShortcut(GlobalShortcut { accelerator }))
+    } else {
+      Err(ShortcutManagerError::InvalidAccelerator(
+        "Unable to register global shortcut".into(),
```

Comment:
Can we make this error more helpful? i.e. why it failed and maybe include a dump of the failed `Accelerator` struct.